api/registrations/views.py

```diff
@@ -540,6 +545,78 @@ def get_queryset(self):
         auth = get_user_auth(self.request)
         return sorted([each for each in nodes if each.can_view(auth)], key=lambda n: n.date_modified, reverse=True)
 
+
+class RegistrationCitationDetail(JSONAPIBaseView, generics.RetrieveAPIView, RegistrationMixin):
+    """ The registration citation for a registration in CSL format *read only*
+
+    ##Note
+    **This API endpoint is under active development, and is subject to change in the future**
+
+    ##RegistraitonCitationDetail Attributes
+
+        name                     type                description
+        =================================================================================
+        id                       string               unique ID for the citation
+        title                    string               title of project or component
+        author                   list                 list of authors for the work
+        publisher                string               publisher - most always 'Open Science Framework'
+        type                     string               type of citation - web
+        doi                      string               doi of the resource
+
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.NODE_REGISTRATIONS_READ]
+    required_write_scopes = [CoreScopes.NULL]
+
+    serializer_class = NodeCitationSerializer
+    view_category = 'registrations'
+    view_name = 'registration-citation'
+
+    def get_object(self):
+        node = self.get_node()
+        return node.csl
+
+
+class RegistrationCitationStyleDetail(JSONAPIBaseView, generics.RetrieveAPIView, RegistrationMixin):
+    """ The registration citation for a registration in a specific style's format t *read only*
+
+        ##Note
+        **This API endpoint is under active development, and is subject to change in the future**
+
+    ##RegistrationCitationStyleDetail Attributes
+
+        name                     type                description
+        =================================================================================
+        citation                string               complete citation for a registration in the given style
+
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.NODE_REGISTRATIONS_READ]
+    required_write_scopes = [CoreScopes.NULL]
+
+    serializer_class = NodeCitationStyleSerializer
+    view_category = 'registrations'
+    view_name = 'registration-style-citation'
+
+    def get_object(self):
```

Comment:
Same here. You could probably save a bit of duplication but subclassing the node version.